lib/routes/home_route/search_route.dart

```diff
@@ -609,82 +592,94 @@ class _SearchRouteState extends State<SearchRoute> with SelectionHandlerMixin {
+                  ),
+                  actionsSelection: selectionRoute
+                    ? [
+                        if (widget.delegate.query.isNotEmpty)
+                          selectAllAction,
+                        if (widget.delegate.query.isNotEmpty)
+                          const _ClearButton(),
                       ]
-                  : [
-                    DeleteSongsAppBarAction<Content>(
-                      controller: stateDelegate.selectionController,
-                    ),
-                    ValueListenableBuilder<Type?>(
-                      valueListenable: stateDelegate.onContentTypeChange,
-                      builder: (context, contentType, child) => AnimatedSwitcher(
-                        duration: const Duration(milliseconds: 400),
-                        switchInCurve: Curves.easeOut,
-                        switchOutCurve: Curves.easeIn,
-                        transitionBuilder: (child, animation) => EmergeAnimation(
-                          animation: animation,
-                          child: child,
+                    : [
+                        ValueListenableBuilder<Type?>(
+                          valueListenable: stateDelegate.onContentTypeChange,
+                          builder: (context, contentType, child) => stateDelegate.contentType == null
+                            ? const SizedBox.square(dimension: NFConstants.iconButtonSize)
+                            : const SizedBox.shrink(),
                         ),
-                        child: stateDelegate.contentType == null
-                          ? const SizedBox.shrink()
-                          : SelectAllSelectionAction<Content>(
-                              controller: selectionController,
-                              entryFactory: (content, index) => SelectionEntry.fromContent(
-                                content: content,
-                                index: index,
-                                context: context,
-                              ),
-                              getAll: () => stateDelegate.results.map.getValue(contentType)!,
-                            ),
-                      ),
-                    ),
-                  ],
-                elevationSelection: 0.0,
-                elevation: theme.appBarTheme.elevation!,
-                toolbarHeight: kToolbarHeight,
-                backgroundColor: theme.primaryColor,
-                iconTheme: theme.primaryIconTheme,
-                textTheme: theme.primaryTextTheme,
-                leading: buildLeading(),
-                actions: selectionRoute ? const [] : buildActions(),
-                bottom: bottom,
-                title: !selectionRoute ? title : const SizedBox.shrink(),
+                        DeleteSongsAppBarAction<Content>(
+                          controller: stateDelegate.selectionController,
+                          keepSpacingWhenHidden: true,
```

Comment:
Deleted

Reply:
I'ts definitely better, but it looks like there is still more spacing in the back than in the front because of the padding here:
https://github.com/nt4f04uNd/sweyer/blob/68c7dd5cb8e3735d0afcfbb6532febaee2a5c622/lib/routes/home_route/search_route.dart#L580
I'm wondering if we still need it, since the `SelectionAppBar` also adds it's own padding.

| Without Padding | ![Without Padding](https://user-images.githubusercontent.com/6966049/172021114-9bc360f3-dde7-4196-833b-1d21b0059b5c.png) |
|-------------------|---------------------|
| With Padding      | ![With Padding](https://user-images.githubusercontent.com/6966049/172021109-3d9b70fa-0b53-4389-8bd6-68e9066c8e5b.png) |
| `1.0.8` branch | ![1 0 8 branch](https://user-images.githubusercontent.com/6966049/172021578-114cb002-a1d4-4b62-9579-7367f5161b0e.png) |

This is not something that was introduced by this pull request, so feel free to ignore this, it's just something I noticed.